File: python/linear_search.py
def linear_search(arr, target):

    for index, value in enumerate(arr):  
        if value == target:
            return index  
    return -1  

# Example
numbers = [10, 20, 30, 40, 50]
target = 30

    for i in range(len(arr)):
        if arr[i] == target:
            return i
    return -1

# Example usage
arr = [4, 2, 7, 1, 9]
target = 7
result = linear_search(arr, target)
print(f"Element found at index: {result}" if result != -1 else "Element not found")

File: php/LinearSearch.php
<?php
function linearSearch($arr, $target) {
    foreach ($arr as $index => $value) {
        if ($value == $target) {
            return $index; // found
        }
    }
    return -1; // not found
}

// Example array
$arr = [12, 34, 54, 2, 3];

// Element to search
$target = 54;

// Perform search
$result = linearSearch($arr, $target);

// Display results in a user-friendly way
echo "Searching for $target in the array: [" . implode(", ", $arr) . "]\n\n";

if ($result != -1) {
    echo "✅ Found $target at position " . ($result + 1) . " (index $result)";
} else {
    echo "❌ Sorry, $target was not found in the array.";
}
?>
